### PR TITLE
reformat dependencies section of application.yml  and service.yml

### PR DESCRIPTION
**Pull request by cheahuychou**
Make the dependencies section of `application.yml` and `service.yml` have the same format to make reusing dependency definitions (https://github.com/Originate/exosphere-sdk/issues/284)  easier. So the dependencies section now looks like:
```
dependencies:
  - mongo:
    name: 'mongo'
    version: '3.4.0'
    config:
      volumes:
        - '{{EXO_DATA_PATH}}:/data/db'
      ports:
        - '27017:27017'
      online-text: 'waiting for connections'
```
instead of
```
dependencies:
  - type: exocom
    version: 0.22.1
``` 
and 
```
dependencies:
  mongo:
    dev:
      image: 'mongo'
      version: '3.4.0'
      volumes:
        - '{{EXO_DATA_PATH}}:/data/db'
      ports:
        - '27017:27017'
      online-text: 'waiting for connections'
```

### CODE REVIEW

File: cli/features/tutorial.feature [hunk 1]
```diff
@@ -162,10 +162,10 @@ Feature: Following the tutorial
           - todo.updated
 
       dependencies:
-        mongo:
-          dev:
-            image: 'mongo'
-            version: '3.4.0'
+        - mongo:
```
Comment (hugoesthere) [inline cli/features/tutorial.feature]: Can we keep this in the same format that it is in `application.yml`?
```
dependencies:
- name: 'mongo'
  version: '3.4.0'
  config:
     ...
```
Looks like you have that format in a couple other places as well, let's stick with the one above.

File: exo-setup/src/docker-setup.ls [hunk 2]
```diff
@@ -32,17 +32,19 @@ class DockerSetup
       links: @_get-docker-links!
       environment: @_get-docker-env-vars!
       depends_on: @_get-service-dependencies!
-    for dependency, dependency-config of @service-config.dependencies
-      docker-config[dependency + dependency-config.dev.version] = @_get-service-dependency-docker-config dependency, dependency-config.dev
+    if @service-config.dependencies
```
Comment (hugoesthere) [inline exo-setup/src/docker-setup.ls]: Is this conditional necessary? The for loop won't execute if there are no dependencies right?

File: exo-setup/src/docker-setup.ls [hunk 3]
```diff
@@ -32,17 +32,19 @@ class DockerSetup
...
     docker-config
 
 
   # compiles list of container links from service to dependency
   # returns undefined if length is 0 so it can be ignored with Obj.compact
   _get-docker-links: ->
     links = []
-    for dependency, dependency-config of @service-config.dependencies
-      links.push "#{dependency + dependency-config.dev.version}:#{dependency}"
+    if @service-config.dependencies
```
Comment (hugoesthere) [inline exo-setup/src/docker-setup.ls]: Same as above.

File: exo-setup/src/docker-setup.ls [hunk 4]
```diff
@@ -53,24 +55,26 @@ class DockerSetup
     for dependency-config in @app-config.dependencies
       dependency = ApplicationDependency.build dependency-config
       env-vars = {...env-vars, ...dependency.get-service-env-variables!}
-    for dependency of @service-config.dependencies
-      env-vars[dependency.to-upper-case!] = dependency
+    if @service-config.dependencies
```
Comment (hugoesthere) [inline exo-setup/src/docker-setup.ls]: Same as above.

File: exo-setup/src/docker-setup.ls [hunk 5]
```diff
@@ -53,24 +55,26 @@ class DockerSetup
...
+        env-vars[dependency.name.to-upper-case!] = dependency.name
     env-vars
 
 
   # compiles list of names of dependencies a service relies on
   _get-service-dependencies: ->
     dependencies = @_get-app-dependencies!
-    for dependency, dependency-config of @service-config.dependencies
-      dependencies.push "#{dependency}#{dependency-config.dev.version}"
+    if @service-config.dependencies
```
Comment (hugoesthere) [inline exo-setup/src/docker-setup.ls]: Same as above.

File: cli/features/tutorial.feature (hunk 1 above)
Comment (cheahuychou) [reply cli/features/tutorial.feature]: Yes, that was a mistake. I'll change it.

File: exo-setup/src/docker-setup.ls (hunk 2 above)
Comment (cheahuychou) [reply exo-setup/src/docker-setup.ls]: I think without that if statement, it returns `Cannot read property 'length' of undefined`

**Comment by cheahuychou** [comment]
@hugobho all comments addressed

File: exo-setup/src/docker-setup.ls (hunk 2 above)
Comment (hugoesthere) [reply exo-setup/src/docker-setup.ls]: What if we do 
```
for dependency in @service-config.dependencies or {}
```
What do you think? I feel like it reads a little nicer.

Comment (cheahuychou) [reply exo-setup/src/docker-setup.ls]: This is an array so should be `for dependency in @service-config.dependencies or []`. I think that works.

File: exo-test/src/service-tester.ls [hunk 6]
```diff
@@ -44,21 +44,22 @@ class ServiceTester extends EventEmitter
...
     @dependencies = []
-    for dependency-name, dependency-config of @service-config.dependencies
-      if dependency-config.dev
-        @dependencies.push do
-          Image: "#{that.image}:#{that.version}"
-          name: "#{@role}-test-#{dependency-name}"
-          HostConfig: @_get-port-mapping that 
-          online-text: that['online-text']
+    if @service-config.dependencies
+      for dependency in @service-config.dependencies
```
Comment (hugoesthere) [inline exo-test/src/service-tester.ls]: Let's do `for dependency in @service-config.dependencies or []` here also

**Review by hugoesthere — APPROVED** [review]
👍

**Pull request closed without merging**